src/Analytics.js

```diff
@@ -123,8 +123,8 @@ const LAST_VISIT_TS_KEY = "mx_Riot_Analytics_lvts";
 
 function getUid() {
     try {
-        let data = localStorage.getItem(UID_KEY);
-        if (!data) {
+        let data = localStorage && localStorage.getItem(UID_KEY);
```

Comment:
evidence from the test environment shows that this doesn't always work and can throw errors. Generally best to `typeof(localStorage) === 'object' && localStorage.getItem(whatever)` instead.

Reply:
Fair enough, this worked in my tests and matches the other null-guards in the app at least.

Reply:
It's possibly only a NodeJS environment and browsers don't fail as annoyingly. https://github.com/matrix-org/matrix-js-sdk/pull/1300 was the recent failure in mind.

Reply:
going to merge as we use the same null-guard in most places. Will keep an eye out for it failing.